xapian-core/include/xapian/cluster.h

```diff
@@ -180,64 +182,153 @@ class XAPIAN_VISIBILITY_DEFAULT TermListGroup : public FreqSource {
     void add_document(const Document &doc);
 
     // Add a number of documents from the DocumentSource
-    void add_documents(DocumentSource &docs);
+    void add_documents(MSetDocumentSource docs);
 
     Xapian::doccount get_termfreq(const std::string &tname);
 
     Xapian::doccount get_doccount();
 };
 
+/** Structure to store term and corresponding wdf.
+ *  This is used with PointTermIterator to return wdf
+ */
+struct XAPIAN_VISIBILITY_DEFAULT Wdf {
```

Comment:
Is this used by clients of Xapian, or just within Xapian? If the latter, it shouldn't be in a public header file. (In which case you might be able to use a `std::pair<>` instead.) I'm not entirely sure that the `Point` stuff needs to be exposed, either.